File: Data Structures & Algorithms/longest-consecutive-sequence/submission-13.java
public class Solution {
    public int longestConsecutive(int[] nums) {
        // numSet being a HashSet will get rid of duplicates, unlike HashMap which uses key-value pairs, HashSet is just values 
        Set<Integer> numSet = new HashSet<>();
        // Populating numSet HashSet
        for (int num : nums) {
            numSet.add(num);
        }
        int longest = 0;

        for (int num : numSet) {
            // The moment we find a number (start num) which has no number smaller than it
            if (!numSet.contains(num - 1)) {
                // lenghth is 1 since we found start and that counts as 1 already
                int length = 1;
                // While there is a sequence 
                while (numSet.contains(num + length)) {
                    length++;   // Increase length
                }
                // We use Math.max() here is because we may find multiple sequences, and we want to keep track of the longest one seen so far.
                // This is useful because there may exist multiple numbers in nums that do not have a number smaller than it by one
                longest = Math.max(longest, length);
            }
        }
        return longest;
    }
}

File: Data Structures & Algorithms/rotate-array/submission-13.cpp
class Solution {
private: 
    void reverse(vector<int> &nums, int left, int right){
        while(left < right){
            swap(nums[left], nums[right]);
            left++;
            right--;
        }
    }

public:
    void rotate(vector<int>& nums, int k) {

        // Have to modulo k by nums.size() because there's no guarantee that k is less than nums.size()!
        k %= nums.size();

        reverse( nums, 0, nums.size() - 1 );
        reverse(nums, 0, k - 1);
        reverse(nums, k, nums.size() - 1 );



    }
};